paddle/fluid/framework/ir/mkldnn/scale_matmul_fuse_pass_tester.cc

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/framework/ir/mkldnn/scale_matmul_fuse_pass.h"
+#include <gtest/gtest.h>
+
+namespace paddle {
+namespace framework {
+namespace ir {
+
+void SetOp(ProgramDesc* prog, const std::string& type, const std::string& name,
+           const std::vector<std::string>& inputs,
+           const std::vector<std::string>& outputs, float scale = 1.0f,
+           float bias = 0.0f) {
+  auto* op = prog->MutableBlock(0)->AppendOp();
+
+  op->SetType(type);
+  if (type == "scale") {
+    op->SetInput("X", {inputs[0]});
+    op->SetAttr("scale", scale);
+    op->SetAttr("bias", bias);
+  } else if (type == "matmul") {
+    op->SetInput("X", {inputs[0]});
+    if (inputs.size() > 1) op->SetInput("Y", {inputs[1]});
+    op->SetAttr("alpha", scale);
+  } else {
+    FAIL() << "Unexpected operator type.";
+  }
+  op->SetOutput("Out", {outputs[0]});
+}
+
+// a->scale->b
+// (b,c)->matmul->d
+ProgramDesc BuildProgramDesc(float scale, float bias, float alpha) {
+  ProgramDesc prog;
+
+  for (auto& v : std::vector<std::string>({"a", "b", "c", "d"})) {
+    prog.MutableBlock(0)->Var(v);
+  }
+
+  SetOp(&prog, "scale", "Scale", {"a"}, {"b"}, scale, bias);
+  SetOp(&prog, "matmul", "Matmul", {"b", "c"}, {"d"}, alpha);
+
+  return prog;
+}
+
+void MainTest(const ProgramDesc& prog, int removed_nodes_count,
+              const std::vector<std::string> scale_in_out,
+              const std::vector<std::string> matmul_in_out, float alpha) {
+  std::unique_ptr<ir::Graph> graph(new ir::Graph(prog));
+  int original_nodes_num = graph->Nodes().size();
+  auto pass = PassRegistry::Instance().Get("scale_matmul_fuse_pass");
+  graph.reset(pass->Apply(graph.release()));
+  int current_nodes_num = graph->Nodes().size();
+
+  for (auto* node : graph->Nodes()) {
+    if (node->IsOp()) {
+      auto* op = node->Op();
+      if (op->Type() == "scale") {
+        EXPECT_EQ(op->Input("X")[0], scale_in_out[0]);
+        EXPECT_EQ(op->Output("Out")[0], scale_in_out[1]);
+      } else if (op->Type() == "matmul") {
+        EXPECT_EQ(op->Input("X")[0], matmul_in_out[0]);
+        EXPECT_EQ(op->Input("Y")[0], matmul_in_out[1]);
+        EXPECT_EQ(op->Output("Out")[0], matmul_in_out[2]);
+        auto alpha = op->GetAttrIfExists<float>("alpha");
+        EXPECT_EQ(op->GetAttrIfExists<float>("alpha"), alpha);
```

Comment:
I think you meant not to include the line 77

Reply:
Done